models/src/main/scala/coop/rchain/models/rholangn/parmanager/protobuf/ProtoCodec.scala

```diff
@@ -0,0 +1,34 @@
+package coop.rchain.models.rholangn.parmanager.protobuf
+
+import cats.Applicative
+import cats.syntax.all._
+import com.google.protobuf.{CodedInputStream, CodedOutputStream}
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
+import scala.util.Using
+
+object ProtoCodec {
+
+  // TODO: Properly handle errors
+  @SuppressWarnings(Array("org.wartremover.warts.TryPartial"))
+  def decode[F[_], T](bv: Array[Byte], read: CodedInputStream => F[T]): F[T] =
+    Using(new ByteArrayInputStream(bv)) { input =>
+      val cis = CodedInputStream.newInstance(input)
+      read(cis)
+    }.get
+
+  // TODO: Properly handle errors
+  @SuppressWarnings(Array("org.wartremover.warts.TryPartial"))
+  def encode[F[_]: Applicative](
+      payloadSize: Int,
+      write: CodedOutputStream => F[Unit]
+  ): F[Array[Byte]] =
+    Using(new ByteArrayOutputStream(payloadSize)) { baos =>
+      val cos = CodedOutputStream.newInstance(baos)
+      write(cos).map { _ =>
+        cos.flush()
+        baos.flush()
+        baos.toByteArray
+      }
+    }.get
```

Comment:
CodedOutputStream is internal to protobuf, caller should not need to know when returning bytes.

We have serializer which needs Writer and Reader so this is what protobuf here should provide.

Reply:
Not sure why caller should not know about CodedOutputStream. This file is only about protobuf and I did not want to pull here anything related to serialisation, or ParN. If I have here only PrimitiveWriter - I have to also pull and instance of ParN that I want to serialise. But this is what should be avoided?